### PR TITLE
Fix require highjack to also work on singleton

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb Ready for review.

File: lib/mutant/require_highjack.rb [hunk 1]
```diff
@@ -10,10 +10,14 @@ module RequireHighjack
     # @return [#call]
     #   the original implementation on singleton
     def self.call(target, callback)
-      target.method(:require).tap do
+      target.public_method(:require).tap do
         target.module_eval do
+          undef_method(:require)
           define_method(:require, &callback)
-          public :require
+          class << self
```
Comment (dkubb) [inline lib/mutant/require_highjack.rb]: Are you able to use `singleton_class.undef_method(:require)` here instead?

Comment (mbj) [inline lib/mutant/require_highjack.rb]: I did not assume this is public. If its public its better to use this form.

Comment (dkubb) [inline lib/mutant/require_highjack.rb]: > I did not assume this is public

I *think* it is. The Ruby documentation doesn't *usually* have any entries for private methods afaik.

Comment (mbj) [inline lib/mutant/require_highjack.rb]: Counter example: 

```
     NoMethodError:
       private method `undef_method' called for #<Class:#<Module:0x0056106133bf90>>
```

Comment (mbj) [inline lib/mutant/require_highjack.rb]: Should we prefer:

`singleton_class.send(:undef_method, :require)` over the `class << self` here?

Comment (dkubb) [inline lib/mutant/require_highjack.rb]: No, I dislike `#send`. At best we could use `singleton_class.instance_eval { undef_method(:require) }`, but that's probably not better than `class << self` which is a built-in language construct.

Comment (mbj) [inline lib/mutant/require_highjack.rb]: Yeah, lets go with the buildin for now.

**Comment by mbj** [comment]
As there are no unaddressed flags left I'll merge this one.

**Pull request merged**